### PR TITLE
PR for issue 27 - adding multiple wildcard support

### PR DESCRIPTION
**Pull request by champagnetony**
Coverage is not 100% but I've had some trouble figuring out how to get the extra branches covered in the multi-wildcard state. Any suggestions would be appreciated. Based on my limited understanding of the code base, this approach does work, but I could see that there could be better ways to tackle this problem in a more programmatically clean design.

### CODE REVIEW

File: lib/modifiers.js [hunk 1]
```diff
@@ -45,7 +45,19 @@ function nestedRestore (arr) {
...
   for (var i = 0; i < length; i++) {
     const { key, target, value } = arr[i]
-    target[key] = value
+    if (has(target, key)) {
+      target[key] = value
+    }
+    /* istanbul ignore else */
+    if (typeof target === 'object') {
+      const targetKeys = Object.keys(target)
+      targetKeys.forEach((tKey) => {
```
Comment (mcollina) [inline lib/modifiers.js:54]: use a for(;;) loop instead, it's way faster.

File: lib/modifiers.js [hunk 2]
```diff
@@ -80,23 +100,56 @@ function specialSet (o, k, path, afterPath, censor, isCensorFct, censorFctTakesP
...
-    n = n[k]
+    if (k === '*') {
+      wc = k
+      if (i !== lastPathIndex) {
+        continue
+      }
+    }
+    if (wc) {
+      const wcKeys = Object.keys(n)
+      wcKeys.forEach((wck) => {
```
Comment (mcollina) [inline lib/modifiers.js:121]: please use a for(;;) loop?

**Review by mcollina — COMMENTED** [review]
Have you run the benchmarks before/after your change?

File: lib/modifiers.js (hunk 1 above)
Comment (champagnetony) [reply lib/modifiers.js:54]: OK :)

**Comment by champagnetony** [comment]
> Have you run the benchmarks before/after your change?

benchNoirV2*500: 65.079ms
benchFastRedact*500: 19.302ms
benchFastRedactRestore*500: 16.967ms
benchNoirV2Wild*500: 32.028ms
benchFastRedactWild*500: 33.008ms
benchFastRedactWildRestore*500: 39.683ms
benchFastRedactIntermediateWild*500: 163.621ms
benchFastRedactIntermediateWildRestore*500: 205.764ms
benchJSONStringify*500: 296.145ms
benchNoirV2Serialize*500: 403.711ms
benchFastRedactSerialize*500: 299.483ms
benchNoirV2WildSerialize*500: 399.529ms
benchFastRedactWildSerialize*500: 323.342ms
benchFastRedactIntermediateWildSerialize*500: 506.828ms
benchFastRedactIntermediateWildMatchWildOutcomeSerialize*500: 808.960ms
benchFastRedactStaticMatchWildOutcomeSerialize*500: 345.606ms
benchNoirV2CensorFunction*500: 26.928ms
benchFastRedactCensorFunction*500: 35.535ms
benchFastRedactCensorFunctionIntermediateWild*500: 199.417ms
benchFastRedactCensorFunctionWithPath*500: 20.742ms
benchFastRedactWildCensorFunctionWithPath*500: 314.464ms
benchFastRedactIntermediateWildCensorFunctionWithPath*500: 136.226ms
benchNoirV2*500: 48.579ms
benchFastRedact*500: 9.105ms
benchFastRedactRestore*500: 9.310ms
benchNoirV2Wild*500: 17.255ms
benchFastRedactWild*500: 27.130ms
benchFastRedactWildRestore*500: 31.067ms
benchFastRedactIntermediateWild*500: 128.558ms
benchFastRedactIntermediateWildRestore*500: 278.786ms
benchJSONStringify*500: 301.368ms
benchNoirV2Serialize*500: 387.137ms
benchFastRedactSerialize*500: 299.639ms
benchNoirV2WildSerialize*500: 390.838ms
benchFastRedactWildSerialize*500: 320.356ms
benchFastRedactIntermediateWildSerialize*500: 524.510ms
benchFastRedactIntermediateWildMatchWildOutcomeSerialize*500: 817.824ms
benchFastRedactStaticMatchWildOutcomeSerialize*500: 350.235ms
benchNoirV2CensorFunction*500: 18.428ms
benchFastRedactCensorFunction*500: 27.464ms
benchFastRedactCensorFunctionIntermediateWild*500: 176.990ms
benchFastRedactCensorFunctionWithPath*500: 17.310ms
benchFastRedactWildCensorFunctionWithPath*500: 313.050ms
benchFastRedactIntermediateWildCensorFunctionWithPath*500: 140.539ms

File: lib/modifiers.js (hunk 1 above)
Comment (champagnetony) [reply lib/modifiers.js]: done

File: lib/modifiers.js (hunk 2 above)
Comment (champagnetony) [reply lib/modifiers.js]: done

**Comment by davidmarkclements** [comment]
@champagnetony if you run `npm run cov-ui` you'll get a visual coverage report, this should make it easier to figure out what tests still need to be written

**Comment by champagnetony** [comment]
> @champagnetony if you run `npm run cov-ui` you'll get a visual coverage report, this should make it easier to figure out what tests still need to be written

Thank you - There are only 3 lines that need the various paths covered, but I just don't know how to exercise those paths. Any suggestions would be welcome.

**Comment by andrhamm** [comment]
@champagnetony Do you expect you'll have any time to complete the test coverage? Multiple wildcard support would be great to have! Thanks

**Comment by jsumners** [comment]
Anyone can pick up the work and retain authorship of prior work -- https://gist.github.com/jsumners/461ef7a64545108635cc437fde112721

**Comment by champagnetony** [comment]
> @champagnetony Do you expect you'll have any time to complete the test coverage? Multiple wildcard support would be great to have! Thanks

I've been trying to get back to it but work has gotten busier and then I've been out of town a lot. If you'd like to finish the tests, please go ahead.

**Comment by lrecknagel** [comment]
I´ve tried my best to finish this PR as advised, see #40 
Happy to improve if still things remain.

**Comment by champagnetony** [comment]
> I´ve tried my best to finish this PR as advised, see #40
> Happy to improve if still things remain.

Thank you for picking this up!

**Pull request merged**